code/cutscene/cutscenes.cpp

```diff
@@ -48,38 +49,101 @@ void cutscene_close()
 		}
 }
 
+static cutscene_info *get_cutscene_pointer(char *cutscene_filename)
+{
+	for (int i = 0; i < Num_cutscenes; i++) {
+		if (!stricmp(cutscene_filename, Cutscenes[i].filename)) {
+			return &Cutscenes[i];
+		}
+	}
+
+	// Didn't find anything.
+	return NULL;
+}
+
+static void cutscene_info_init(cutscene_info *csni)
+{
+	csni->filename[0] = '\0';
+	csni->name[0] = '\0';
+	csni->description = NULL;
+	csni->flags = {};
+}
+
 // initialization stuff for cutscenes
-void cutscene_init()
+void parse_cutscene_table(const char* filename)
 {
-	atexit(cutscene_close);
+
 	char buf[MULTITEXT_LENGTH];
-	cutscene_info cutinfo;
+	cutscene_info csnt;
+	cutscene_info_init(&csnt);
+
+	cutscene_info *csnp;
+	bool create_if_not_found = true;
 
 	try
 	{
-		read_file_text("cutscenes.tbl", CF_TYPE_TABLES);
+		
+		read_file_text(filename, CF_TYPE_TABLES);
+
 		reset_parse();
 
 		// parse in all the cutscenes
-		Cutscenes.clear();
 		skip_to_string("#Cutscenes");
 		ignore_white_space();
 
-		bool isFirstCutscene = true;
-
 		while (required_string_either("#End", "$Filename:"))
 		{
 			required_string("$Filename:");
-			stuff_string(cutinfo.filename, F_PATHNAME, MAX_FILENAME_LEN);
+			stuff_string(csnt.filename, F_PATHNAME, MAX_FILENAME_LEN);
 
-			required_string("$Name:");
-			stuff_string(cutinfo.name, F_NAME, NAME_LENGTH);
+			if (optional_string("+nocreate")) {
+				if (!Parsing_modular_table) {
+					Warning(LOCATION, "+nocreate flag used for cutscene in non-modular table\n");
+				}
+				create_if_not_found = false;
+			}
+
+			// Does this cutscene exist already?
+			// If so, load this new info into it
+			csnp = get_cutscene_pointer(csnt.filename);
+			if (csnp != NULL) {
+				if (Parsing_modular_table) {
+					nprintf(
+						("Warning", "More than one version of cutscene %s exists; using newer version.\n", csnt.filename));
+				} else {
+					error_display(1,
+						"Error:  Cutscene %s already exists.  All cutscene names must be unique.",
+						csnt.filename);
+				}
+			} else {
+				// Don't create cutscene if it has +nocreate and is in a modular table.
+				if (!create_if_not_found && Parsing_modular_table) {
+					if (!skip_to_start_of_string_either("$Fileame:", "#End")) {
+						error_display(1, "Missing [#End] or [$Filename] after cutscene %s", csnt.filename);
+					}
+					return;
+				}
+				Cutscenes.push_back(csnt);
+				csnp = &Cutscenes[Num_cutscenes++];
+			}
 
-			required_string("$Description:");
-			stuff_string(buf, F_MULTITEXT, sizeof(buf));
-			drop_white_space(buf);
-			compact_multitext_string(buf);
-			cutinfo.description = vm_strdup(buf);
+			//If $Name is not found use the $Filename instead because this field needs data
+			//but only do this if if we're parsing the tbl and not a tbm
+			if (optional_string("$Name:")) {
+				stuff_string(csnp->name, F_NAME, NAME_LENGTH);
+				mprintf(("Adding cutscene %s\n", csnp->name));
+			} else if (!Parsing_modular_table) {
+				strncpy(csnp->name, csnt.filename, 32);
```

Comment:
`strncpy` is a bit messy to use, and this would have to be tweaked in a couple of ways, but fortunately it isn't even needed here.  Replace this line with
```
strcpy_s(csnp->name, csnt.filename);
```